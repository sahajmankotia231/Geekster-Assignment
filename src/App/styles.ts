import styled from "styled-components";

import { BLACK } from "constants/colors";
import { ReactComponent as LogoSvg } from "assets/logo.svg";
import { ReactComponent as PartnersSvg } from "assets/partners.svg";

export const AppContainer = styled.div`
  width: 100%;
  height: 100vh;
`;

export const Content = styled.div`
  padding: 16px;
`;

export const Logo = styled(LogoSvg)`
  width: 120px;
  height: 32px;
  margin-bottom: 86px;
`;

export const Partners = styled(PartnersSvg)`
  height: 19px;
  margin-top: 32px;
`;

export const Title = styled.h2`
  color: ${BLACK};
  font-size: 26px;
  font-weight: normal;
  line-height: 36px;
  margin-bottom: 146px;
`;

export const FieldSet = styled.fieldset`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-bottom: 12px;
`;

export const GuestsContainer = styled.div`
  width: 80px;
  max-width: 80px;
  margin-left: 12px;
`;
