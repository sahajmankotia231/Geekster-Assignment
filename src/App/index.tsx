import React, { ReactElement, useState } from "react";
import { Provider } from "react-redux";

import store from "store";
import Guests from "Guests";

import { AppContainer, Title, Logo, Content, Partners } from "./styles";
import AppForm from "./components/AppForm";

const App = (): ReactElement => {
  const [isGuestsVisible, setIsGuestsVisible] = useState(false);

  return (
    <Provider store={store}>
      <AppContainer id="root">
        <Content>
          <Logo />
          <Title>
            Find the perfect <br /> deal, always.
          </Title>
          <AppForm openGuests={() => setIsGuestsVisible(true)} />
          <Partners />
          {isGuestsVisible && (
            <Guests onClose={() => setIsGuestsVisible(false)} />
          )}
        </Content>
      </AppContainer>
    </Provider>
  );
};

export default App;
