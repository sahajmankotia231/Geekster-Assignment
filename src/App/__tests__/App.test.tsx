import React from "react";
import { render } from "@testing-library/react";

import App from "..";

describe("<App />", () => {
  it("should render properly", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
