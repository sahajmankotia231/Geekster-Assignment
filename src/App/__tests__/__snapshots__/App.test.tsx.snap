// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render properly 1`] = `
<div
  class="sc-hBURRC fPbypC"
  id="root"
>
  <div
    class="sc-fotPbf edKkdo"
  >
    <svg
      class="sc-fFehDp gOYNzr"
    >
      logo.svg
    </svg>
    <h2
      class="sc-ieebsP cVEnuv"
    >
      Find the perfect 
      <br />
       deal, always.
    </h2>
    <div
      class="sc-dlVyqM fPlOTZ"
    >
      <form>
        <fieldset
          class="sc-dJjZJu cMwvGz"
        >
          <div
            class="sc-kfPtPH dYvmmi"
          >
            <i
              class="sc-fKVsgm pSvAj"
            >
              <svg>
                pin.svg
              </svg>
            </i>
            <input
              class="sc-iwjezw kwSriW"
              placeholder="Type city, place, or hotel name"
              type="text"
              value=""
            />
            <i
              class="sc-bBHwJV hEXRLh"
            >
              <svg>
                cursor.svg
              </svg>
            </i>
          </div>
        </fieldset>
        <fieldset
          class="sc-dJjZJu cMwvGz"
        >
          <div
            class="sc-cxpRKc bhQfma"
          >
            <svg>
              calendar.svg
            </svg>
            <strong>
              Check-in
            </strong>
            <span
              class="sc-llYToB fxYTXF"
            />
            <strong>
              Check-out
            </strong>
          </div>
          <div
            class="sc-hGPAah HFxPm"
          >
            <div
              class="sc-kfPtPH dYvmmi"
            >
              <i
                class="sc-fKVsgm pSvAj"
              >
                <svg>
                  guests.svg
                </svg>
              </i>
              <input
                class="sc-iwjezw jNduHP"
                max="9"
                min="1"
                type="number"
                value="1"
              />
            </div>
          </div>
        </fieldset>
        <button
          class="sc-bdvvaa tPxtg"
          data-testid="button"
        >
          Search
        </button>
      </form>
    </div>
    <svg
      class="sc-bkkfTU cCuNEv"
    >
      partners.svg
    </svg>
  </div>
</div>
`;
